spec/kitchen/driver/aws/standard_platform_spec.rb

```diff
@@ -0,0 +1,110 @@
+
+require "kitchen/driver/aws/standard_platform"
+require "kitchen/driver/ec2"
+
+describe Kitchen::Driver::Aws::StandardPlatform do
+
+  let(:klass) { Kitchen::Driver::Aws::StandardPlatform }
+
+  describe "parse_platform_string" do
+    let(:platform_string) { "ubuntu-1604-x86_64" }
+
+    it "successfully parses platform, version and architecture" do
+      expect(klass.parse_platform_string(platform_string)).to eq(%w{ubuntu 1604 x86_64})
+    end
+
+    context "when there is just a platform" do
+      let(:platform_string) { "ubuntu" }
+
+      it "successfully parses platform" do
+        expect(klass.parse_platform_string(platform_string)).to eq(["ubuntu", nil, nil])
+      end
+    end
+
+    context "when there is just platform and version" do
+      let(:platform_string) { "ubuntu-1604" }
+
+      it "successfully parses platform and version" do
+        expect(klass.parse_platform_string(platform_string)).to eq(["ubuntu", "1604", nil])
+      end
+    end
+
+    context "when there is an unknown architecture" do
+      let(:platform_string) { "ubuntu-1604-powerpc" }
+
+      it "successfully parses platform and includes architecture in version" do
+        expect(klass.parse_platform_string(platform_string)).to eq(["ubuntu", "1604-powerpc", nil])
+      end
+    end
+  end
+
+  describe "from_platform_string" do
+    let(:driver) { instance_double(Kitchen::Driver::Ec2) }
+    let(:platform_string) { "ubuntu-1604-x86_64" }
+
+    it "returns an ubuntu StandardPlatform" do
+      expect(klass.from_platform_string(driver, platform_string)).to be_instance_of(klass::Ubuntu)
+    end
+  end
+
+  describe "#sort_images" do
+    let(:img1) { instance_double(::Aws::EC2::Image,
+      creation_date: "1543439623",
+      architecture: "amd64",
+      block_device_mappings: [],
+      root_device_type: "other",
+      virtualization_type: "other",
+      name: "ubuntu"
+    ) }
+    let(:img2) { instance_double(::Aws::EC2::Image,
+      creation_date: "1543439623",
+      architecture: "x86_64",
+      block_device_mappings: [],
+      root_device_type: "other",
+      virtualization_type: "other",
+      name: "ubuntu"
+    ) }
+    let(:img3) { instance_double(::Aws::EC2::Image,
+      creation_date: "1543439623",
+      architecture: "x86_64",
+      block_device_mappings: [],
+      root_device_type: "ebs",
+      virtualization_type: "other",
+      name: "ubuntu"
+    ) }
+    let(:img4) { instance_double(::Aws::EC2::Image,
+      creation_date: "1543439623",
+      architecture: "x86_64",
+      block_device_mappings: [],
+      root_device_type: "ebs",
+      virtualization_type: "hvm",
+      name: "ubuntu"
+    ) }
+    let(:images) { [img1, img2, img3, img4] }
+    let(:sorted_images) { [img4, img3, img2, img1] }
+    let(:standard_platform) { klass.new(nil, nil, nil, nil) }
+
+    it "correctly sorts the images" do
+      expect(standard_platform.send(:sort_images, images)).to eq(sorted_images)
+    end
```

Comment:
![sweet](https://user-images.githubusercontent.com/517302/49184847-5d0d5480-f32e-11e8-8d67-7375e24cc30b.gif)

✨ It's a start! ✨